File: 01sort/sort.cpp
#include <bits/stdc++.h>

using namespace std;
bool sortbyframeno(const pair<int, int> &a, const pair<int, int> &b)
{
    return (a.first < b.first);
}
int main()
{
    vector<pair<int, int>> vect;
    int n;
    cout << "Enter the number of frames" << endl;
    cin >> n;
    for (int i = 0; i < n; i++)
    {
        int x = rand() % 100;
        int data;
        cout << "Enter the data of the frame(" << x << "):" << endl;
        cin >> data;
        vect.push_back({x, data});
    }
    cout << "Before sorting frames:" << endl;
    for (int i = 0; i < vect.size(); i++)
    {
        cout << "Frame" << vect[i].first << ": data is: " << vect[i].second << endl;
    }
    sort(vect.begin(), vect.end(), sortbyframeno);
    cout << "After sorting frames:" << endl;
    for (int i = 0; i < vect.size(); i++)
    {
        cout << "Frame" << vect[i].first << ": data is: " << vect[i].second << endl;
    }
    return 0;
}

File: 01sort/sort.java
import java.util.ArrayList;
import java.util.Scanner;

public class sort {
    static class Frame {
        int frameNo;
        int data;

        public Frame(int frameNo, int data) {
            this.frameNo = frameNo;
            this.data = data;
        }
    }

    public static void main(String[] args) {
        ArrayList<Frame> frameList = new ArrayList<>();

        Scanner scanner = new Scanner(System.in);

        System.out.println("Enter the number of frames:");
        int n = scanner.nextInt();

        for (int i = 0; i < n; i++) {
            int x = (int) (Math.random() * 100);
            System.out.println("Enter the data of the frame(" + x + "):");
            int data = scanner.nextInt();
            frameList.add(new Frame(x, data));
        }

        System.out.println("Before sorting frames:");
        for (Frame frame : frameList) {
            System.out.println("Frame" + frame.frameNo + ": data is: " + frame.data);
        }

        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - 1 - i; j++) {
                if (frameList.get(j).frameNo > frameList.get(j + 1).frameNo) {
                    Frame temp = frameList.get(j);
                    frameList.set(j, frameList.get(j + 1));
                    frameList.set(j + 1, temp);
                }
            }
        }

        System.out.println("After sorting frames:");
        for (Frame frame : frameList) {
            System.out.println("Frame" + frame.frameNo + ": data is: " + frame.data);
        }

        scanner.close();
    }
}
